kobuki_dock_drive/src/dock_drive_states.cpp

```diff
@@ -252,9 +252,9 @@ namespace kobuki {
     // unsigned char right = signal_filt[0];
     unsigned char mid   = signal_filt[1];
     // unsigned char left  = signal_filt[2];
-    RobotDockingState::State next_state = nstate;
```

Comment:
What fix was this making?

This one might be copying state from the passed nstate. I'd have to dig deeper to check. Did you look into that and is that the motivation for this fix?

Also odd that it copies out and copies back instead of working through on the original value. Again, would have to dig deeper.

Reply:
Indeed, it's just an enum.